esdt-safe/src/from_sovereign/token_mapping.rs

```diff
@@ -28,28 +46,31 @@ pub trait TokenMappingModule:
         );
 
         let issue_cost = self.call_value().egld_value().clone_value();
-
-        require!(issue_cost == DEFAULT_ISSUE_COST, "eGLD value should be 0.5");
+        require!(
+            issue_cost == DEFAULT_ISSUE_COST,
+            "eGLD value should be 0.05"
+        );
 
         match token_type {
             EsdtTokenType::Invalid => sc_panic!("Invalid type"),
-            EsdtTokenType::Fungible => self.handle_fungible_token_type(
-                sov_token_id.clone(),
+            EsdtTokenType::Fungible => self.handle_fungible_token_type(FungibleTokenArgs {
+                sov_token_id: sov_token_id.clone(),
                 issue_cost,
                 token_display_name,
                 token_ticker,
                 num_decimals,
-            ),
-            _ => self.handle_nonfungible_token_type(
-                sov_token_id.clone(),
+            }),
+            _ => self.handle_nonfungible_token_type(NonFungibleTokenArgs {
+                sov_token_id: sov_token_id.clone(),
                 token_type,
                 issue_cost,
                 token_display_name,
                 token_ticker,
                 num_decimals,
-            ),
+            }),
         }
 
+        // !!! Unreachable code !!!
```

Comment:
Really? This worked before :thinking:

Reply:
I was thinking the same thing

Reply:
does it really matter if we will probably won't use `esdt-safe`?

Reply:
Currently this contact is used for public deployment. Hopefully soon we'll move to enshrine-esdt.

Reply:
Yep. Issue-ing a token kills current contract execution.